wcomponents-core/src/main/java/com/github/bordertech/wcomponents/render/webxml/WTableRenderer.java

```diff
@@ -301,11 +301,13 @@ private void paintTableActions(final WTable table, final WebXmlRenderContext ren
 						int minRows = constraint.getMinSelectedRowCount();
 						int maxRows = constraint.getMaxSelectedRowCount();
 						String message = constraint.getMessage();
+						int selectedOnOther = table.getSelectedRows().size();
```

Comment:
This looks like it gets the TOTAL selected row count... Shouldn't it only get the OTHER selected row count (i.e. the count of rows selected on all pages except the current one)?

Assuming this code needs to be changed then clearly this has not been adequately unit tested. I suggest you adopt a test-driven approach here - first write the unit test (which should fail) and then make it work.

Reply:
I chose this way as the value would only update when a request is made back to the server. For example, it would start with 0 initially, I would select say 3 items, click the next page, and then it would update with 3. This seemed to work with adding selections, but I didn't take into account going back to a previous page and removing a selection. We want it to instantly update (eg. a button greyed out to normal again) but realised that the way I did it it would only update on refresh/request.

Looking back on it this is a terrible way to do it, and I clearly need to implement more testing. Going to look into different solutions, may possibly go the route we discussed.